CMakeLists.txt

```diff
@@ -18,16 +18,18 @@ macro(enable_strict_compiler_warnings target)
   if (MSVC)
     target_compile_options(${target} PRIVATE /WX /W4)
   elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    target_compile_options(${target} PRIVATE -Wall -Wextra -Wpedantic -Werror -Wold-style-cast -Wfloat-equal -Wmost -Wunused-exception-parameter)
+    target_compile_options(${target} PRIVATE -Wall -Wextra -Wpedantic -Werror -Wfloat-equal -Wmost -Wunused-exception-parameter)
```

Comment:
Why do you remove `-Wold-style-cast` here? Because it causes warnings in header files stemming from websocketpp/asio? I think the better approach is to make these header files being treated as system includes